### PR TITLE
 Not allow, Footer collection venue opening times names dynamic.

### PR DESCRIPTION
**Pull request by ishouty**
*  Not allow, Footer collection venue opening times names dynamic.
*  Replace name and using id to find openingTimes on whats-on.

@jamesgorrie @gestchild @davidpmccormick  - I have done some regression, seems to be all rendering fine. 
* visit-us 
* footer page
* opening-times page
Let me know if you think i would need to introduce any unit testing in any parts of the code I have implemented.

Effected pages: 

- VisitUsStaticContent
- Footer.js 
- whats-on.js 
- VenueClosedPeriods

### CODE REVIEW

**Comment by kenoir** [comment]
@ishouty You can use [draft PRs](https://github.blog/2019-02-14-introducing-draft-pull-requests/) if a PR is WIP.

**Review by jamesgorrie — APPROVED** [review]
Great catch

File: common/model/opening-hours.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+export type OverrideType =
+  | 'Bank holiday'
+  | 'Easter'
+  | 'Christmas and New Year'
+  | 'Late Spectacluar'
+  | 'other';
+
+export type OverrideDate = {
+  overrideDate: Moment,
+  overrideType: OverrideType,
```
Comment (davidpmccormick) [inline common/model/opening-hours.ts:16]: [This should be nullable](https://github.com/wellcomecollection/wellcomecollection.org/blob/master/common/model/opening-hours.js#L17).

Flow lets you have optional keys and nullable values based on which side of the colon the question mark goes. TypeScript only has optional keys (question mark before the colon), so to make something nullable, you have to declare null as a possible value, e.g.

`overrideType: OverrideType | null`

Looks like there are a few other places this'll need updating in this file.

Comment (ishouty) [reply common/model/opening-hours.ts]: Thank you !

**Review by davidpmccormick — APPROVED** [review]
All looks good. Nice.

**Review by davidpmccormick — APPROVED** [review]
🚀

**Pull request merged**